pkg/controller/config/config_controller_test.go

```diff
@@ -417,8 +421,16 @@ func TestConfig_CacheContents(t *testing.T) {
 
 	// Create the Config object and expect the Reconcile to be created
 	ctx = context.Background()
-	err = c.Create(ctx, instance)
-	g.Expect(err).NotTo(gomega.HaveOccurred())
+
+	createInstance := func() client.Object {
+		return configFor([]schema.GroupVersionKind{nsGVK, configMapGVK})
+	}
+
+	// Since we're reusing instance between tests, we must wait for it to be fully
```

Comment:
And this is because of the owner references?  A bit more detail might make this easier for me to understand why this is important

Reply:
This is unrelated to owner references as far as I can tell. I just saw it was flaky, so I fixed it.

Reply:
Thanks for fixing! +1 to Julian's idea of just calling `ensureCreated(instance.DeepCopy())`

Reply:
The contents of the other discussion thread would make a good addition to this comment.  

Also, I believe it's not the client that's re-used, but rather that an dummy API server is started for the entire test suite (see [config_controller_suite_test.go](https://github.com/open-policy-agent/gatekeeper/blob/master/pkg/controller/config/config_controller_suite_test.go#L52)).  

My version of the comment, which you can use as you see fit:

```
// This package uses the same API server process across multiple test functions.
// The residual state from a previous test function can cause flakes.
//
// To ensure a clean slate, we must verify that any previously applied Config object
// has been fully removed before applying our new object.
```

Reply:
> Also, I believe it's not the client that's re-used, but a dummy API server that's started for the entire test suite (see config_controller_suite_test.go).

+1, IIRC, the tests spin up a barebones k8s control plane

Reply:
Done.